### ISSUE TITLE
miou calculate

### ISSUE THREAD

**Issue by cuishuhao** [jfzhang95/pytorch-deeplab-xception#16]
https://github.com/jfzhang95/pytorch-deeplab-xception/blob/6f5a32978bfdb72d7673f77b69ffefc2ae0edb78/dataloaders/utils.py#L163

i think i find something wrong here.
the variable 'unique_label' should contain not only labels of ground truth but also labels of predicts

**Comment by FantasyJXF** [comment]
The new mIoU calculation seems to cheat on performance.

When you generate the confusion matrix like this:
```
    def _generate_matrix(self, gt_image, pre_image):
        mask = (gt_image >= 0) & (gt_image < self.num_class) # The classes not include the white border
        # **gt_image[mask]** change the MATRIX gt_image to a LIST where the *mask*  value is TRUE.
        label = self.num_class * gt_image[mask].astype('int') + pre_image[mask]
        count = np.bincount(label, minlength=self.num_class**2)
        confusion_matrix = count.reshape(self.num_class, self.num_class)
        return confusion_matrix
```

It looks like that you ignore the border of object might cause error, I'm not sure whether it's OK to just remove that part.

**Comment by FantasyJXF** [comment]
```
label = self.num_class * gt_image[mask].astype('int') + pre_image[mask]
```

This line, would you please explain what does this mean? @jfzhang95
